compiler/rustc_hir_analysis/messages.ftl

```diff
@@ -351,7 +351,7 @@ hir_analysis_only_current_traits_arbitrary = only traits defined in the current
 
 hir_analysis_only_current_traits_foreign = this is not defined in the current crate because this is a foreign trait
 
-hir_analysis_only_current_traits_label = impl doesn't use only types from inside the current crate
+hir_analysis_only_current_traits_label = impl doesn't use any uncovered types from inside the current crate
```

Comment:
this is also subtly wrong

the requirement is that given an impl `impl<T0..TN> Trait<P1..Pn> for P0`, there exists an uncovered local type at position `Pi` such that `P0..Pi` does not contain any uncovered generic parameters

`impl<T> Trait<Local> for T {}` also results in this error

Reply:
That's correct of course, I didn't want to increase the verbosity too much.

What do you think of appending "before any uncovered generic parameters". This again isn't totally bullet proof as `P0` is only logically before the other `Pn` but is not before them in the source.

Reply:
maybe something like:
- "note: impl has uncovered type parameters before any local type"
- "note: for more information see <link to reference>"

Reply:
I think it's not so clear that *"no local types"* is also problematic when inverted like thisk. I prefer the original negated version. I do like the shortened *"local type"* instead of *"types from inside the current crate"*, when we link to the reference anyways I think we can get away with it as it expands on what a *"local type"* is: 

- "note: impl doesn't have any local type before any uncovered type parameters"
-  "note: for more information see https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules"

I'm guessing that's the link you meant to add here? Is it ok to link to a fragment of the reference?

I'll see if I can figure out how to add multi line notes (multiple notes?).

Reply:
mutliple notes, yeah

your suggestion looks good to me